frontend/class-frontend.php

```diff
@@ -752,12 +752,16 @@ public function robots() {
 				if ( WPSEO_Options::get( 'noindex-author-wpseo', false ) ) {
 					$robots['index'] = 'noindex';
 				}
-				$curauth = $wp_query->get_queried_object();
-				if ( WPSEO_Options::get( 'noindex-author-noposts-wpseo', false ) && count_user_posts( $curauth->ID, 'any' ) === 0 ) {
-					$robots['index'] = 'noindex';
-				}
-				if ( get_user_meta( $curauth->ID, 'wpseo_noindex_author', true ) === 'on' ) {
-					$robots['index'] = 'noindex';
+
+				if ( $curauth = $wp_query->get_queried_object() ) {
```

Comment:
Assignment in `if-statements` are discouraged. Please move the assignment above the `if` statement and in check if the object has an ID: `property_exists( 'ID', $curauth )`

Reply:
Thanks for your review
Please do it by yourself,  This little tiny code cause problem in our client's websites.

Regards.

Reply:
@stodorovic Working for me..! +1

Reply:
I just tested your commit, but it still doesn't work!

This is weird because in author page the get_queried_object_id() function should return the author ID but it returns zero.

Please use 
```php 
$wp_query->get('author');
```

instead of get_queried_object_id, That works perfectly fine for me.

Reply:
I have tried 

```php
$author_id = $wp_query->get_queried_object_id();
```

and 
```php
$author_id = $wp_query->get('author');
```

Both working fine for me. Not sure why.

Reply:
We could replace `get_queried_object_id()` with `get_query_var`:

```php
$author_id = (int) get_query_var( 'author' );
```
Is it working?

Reply:
Yes, that works fine.